microcosm_elasticsearch/models.py

```diff
@@ -3,17 +3,55 @@
 
 """
 from elasticsearch_dsl import Date, DocType, Keyword
+from elasticsearch_dsl.document import DocTypeOptions, DocTypeMeta
 
 
-class Model(DocType):
+class ModelMeta(DocTypeMeta):
```

Comment:
Can you please add a doc string explaining the purpose of the meta class?

Also: can we somehow avoid having the options (value-type) modify `attrs`? I'm not clear on all of what's happening here, but I really don't understand why `attrs` is shared.

Reply:
It probably is pretty unclear.
I'm not actually changing what the metaclass does - this is the same as what `elasticsearch_dsl` does (see https://github.com/elastic/elasticsearch-dsl-py/blob/master/elasticsearch_dsl/document.py#L25m in particular the `attrs['_doc_type'] = ...` bit comes from there); the actual change is dynamically setting a `Model` field to hold the document's type. The name of the field is defined by the model's `__doctype_field__` attribute.

Reply:
Thanks.